src/cswinrt/strings/WinRT.cs

```diff
@@ -107,7 +107,7 @@ internal class DllModule
         readonly DllGetActivationFactory _GetActivationFactory;
         readonly DllCanUnloadNow _CanUnloadNow; // TODO: Eventually periodically call
 
-        static readonly string _currentModuleDirectory = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+        static readonly string _currentModuleDirectory = AppContext.BaseDirectory;
```

Comment:
I'm not fully sure if this is the right fix, in single-file mode the AppContext.BaseDirectory will point to the location of the single-file executable. Depending on how the `_currentModuleDirectory` is used later, the code could break which is why I opened the issue. If someone uses the `_currentModuleDirectory` to find stuff that got bundled in the `exe` that is going to break 
cc @agocke

Reply:
The only usage for that variable is here
https://github.com/microsoft/CsWinRT/blob/755b8a77f32c566bbca7848cdb0275d915ad7bf0/src/cswinrt/strings/WinRT.cs#L133-L135

Which is just used to get `DllGetActivationFactory` from DLL. That's for fallback, if `RoGetActivationFactory` cannot find factory for the class. And you are prbably right, I'm too fast to apply fix here.

Do I properly understand that `DllGetActivationFactory` is created in DLL by WinRT-component vendor. In case if CsWinRT was used it is comes from here 
https://github.com/microsoft/CsWinRT/blob/ad6adcb997b40cea649cf4156451316e714bc3ea/src/Authoring/WinRT.Host/WinRT.Host.cpp#L300-L303

Short glance at the code indicates that it does not take into account single-file deployment at all. Do I see things correctly?

Reply:
My understanding is that the code is loading a native dll from that path. Native dlls don't get bundled by default, so using the BaseDirectory will work in that case. Unfortunately if somebody specifies `IncludeNativeLibrariesForSelfExtract` the native dll will be bundled and then extracted into a "Temp" directory. But currently we don't provide a good way to find that directory - see https://github.com/dotnet/runtime/issues/43010.

I think one of the native library APIs should be able to try to load the library by a simple name (the same way `DllImport` does), which will find the library always, since the host correctly sets up `NATIVE_DLL_SEARCH_DIRECTORIES`. @elinor-fung Do you know which of the `NativeLibrary` APIs (if any) will trigger the normal runtime's "load library" behavior.

Reply:
[`NativeLibrary.Load(string, Assembly, DllImportSearchPath?)`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.nativelibrary.load#System_Runtime_InteropServices_NativeLibrary_Load_System_String_System_Reflection_Assembly_System_Nullable_System_Runtime_InteropServices_DllImportSearchPath__)

Reply:
I probably misled people a bit here by not showing subsequent code, with assumption that it would be inspected. See below.

https://github.com/microsoft/CsWinRT/blob/755b8a77f32c566bbca7848cdb0275d915ad7bf0/src/cswinrt/strings/WinRT.cs#L133-L141

shows that `Platform.LoadLibraryExW` used primarily for `NETSTANDARD2.0` or `NETCOREAPP2.0`, when on more modern platforms subsequently used `NativeLibrary.TryLoad` which is essentially same as what @elinor-fung proposed. If `dllImportSearchPath == null` trigger use `NATIVE_DLL_SEARCH_DIRECTORIES` then probably that's enough.

Only case which I think is left, is when somehow somebody would like to have "out-of-process" hosting with single-file package. That's `WinRT.Host.cpp` file is for which I mention in previous comments. Is this supported scenario?

Reply:
Single-file is only supported for applications (exe), I would not expect out-of-proc hosted components to be built as applications (similar to COM host components, those are also classlib projects not apps). So that should already disqualify almost all such cases.

On top of that, hosting APIs (hostfxr) don't support single-file in any form currently, so even if the SDK would somehow build it, it won't run.